src/main/java/de/tum/in/www1/artemis/web/rest/lecture/AttachmentUnitResource.java

```diff
@@ -231,4 +284,32 @@ private void checkAttachmentUnitCourseAndLecture(AttachmentUnit attachmentUnit,
             throw new ConflictException("Requested lecture unit is not part of the specified lecture", "AttachmentUnit", "lectureIdMismatch");
         }
     }
+
+    /**
+     * Checks that the lecture exists and is part of a course, and that the user is at least editor in the course
+     *
+     * @param lectureId The id of the lecture
+     */
+    public void checkLecture(Long lectureId) {
+        Lecture lecture = lectureRepository.findByIdWithLectureUnitsElseThrow(lectureId);
+        if (lecture.getCourse() == null) {
+            throw new ConflictException("Specified lecture is not part of a course", "AttachmentUnit", "courseMissing");
+        }
+        authorizationCheckService.checkHasAtLeastRoleInCourseElseThrow(Role.EDITOR, lecture.getCourse(), null);
+    }
+
+    /**
+     * Checks the file exists on the server and is a .pdf
+     *
+     * @param filename the name of the file
+     */
+    public void checkFile(String filename) {
```

Comment:
```suggestion
    private void checkFile(String filename) {
```
Otherwise put it in a service